flax/core/scope.py

```diff
@@ -71,9 +72,26 @@ class DenyList:
 FrozenVariableDict = FrozenDict[str, Collection]
 MutableVariableDict = Dict[str, MutableCollection]
 
+PRNGFoldable = Union[int, str]
 
-def _fold_in_str(rng: PRNGKey, data: str) -> PRNGKey:
-  """Folds a string into a jax.random.PRNGKey using its SHA-1 hash.
+
+class LazyRng(struct.PyTreeNode):
+  rng: PRNGKey
+  suffix: Tuple[Any, ...] = struct.field(pytree_node=False)
```

Comment:
Nit: Shouldn't this have type `Iterable[PRNGFoldable]`, matching the type of `_fold_in_static`'s second argument?

Reply:
It needs to be a tuple to make concat easy but Any should be PRNGFoldable indeed